### DISCUSSION TITLE
redis.exceptions.ConnectionError: Error 32 while writing to socket. Broken pipe.

### DISCUSSION THREAD

**Discussion by harshita01398** [celery/kombu#1565] [Q&A]
To support python 3.7, we had to upgrade to celery 4.3 and we started seeing this exception- redis.exceptions.ConnectionError: Error 32 while writing to socket. Broken pipe.

Celery: 4.3.1
Redis: 4.1.2
Kombu: 4.6.11

After adding health_check_interval, errors reduced but still seeing them in large numbers.

Have already added below setting
broker_transport_options = {
'health_check_interval': 5,
}

Do we have a fix in any latest updates of celery/kombu?

Please suggest a fix on this as it is affecting our production workflows.

[a comment by an automated account is not shown]

**Reply by auvipy** [reply]
I would say upgrade to a supported version first. which is 5.2.x, and for 4.x series try atleast 4.4.6 first.

**Reply by auvipy** [reply]
here is your possible answer https://github.com/celery/kombu/issues/1019#issuecomment-1163138146
